cmd/signature-v4-utils.go

```diff
@@ -260,3 +260,28 @@ func signV4TrimAll(input string) string {
 	// unicode.IsSpace() internally here) to one space and return
 	return strings.Join(strings.Fields(input), " ")
 }
+
+// checkMetaHeaders will check if the metadata from header/url is the same with the one from signed headers
+func checkMetaHeaders(signedHeadersMap http.Header, r *http.Request) APIErrorCode {
+	// check values from http header
+	for k, val := range r.Header {
+		if stringsHasPrefixFold(k, "X-Amz-Meta-") {
+			if signedHeadersMap.Get(k) == val[0] {
+				continue
+			}
+			return ErrUnsignedHeaders
+		}
+	}
+
+	// check values from url, if no http header
+	for k, val := range r.Form {
+		if stringsHasPrefixFold(k, "x-amz-meta-") {
```

Comment:
```suggestion
		if stringsHasPrefixFold(k, "X-Amz-Meta-") {
```

Reply:
@jiuker We want the most likely case (since it is slightly faster). I believe in forms, the most likely case is lower case, while headers are header capitalized.

So AFAICT this matches the capitalization we expect.

Reply:
it is ok to let it like this? I know that the url values have the prefix like this `x-amz-meta-` and I saw that the `stringsHasPrefixFold` function is ignoring case. I would like to leave them like this so that someone knows when they look where the value comes from.